jetty-ee10/jetty-ee10-cdi/src/main/java/org/eclipse/jetty/ee10/cdi/CdiSpiDecorator.java

```diff
@@ -73,21 +77,23 @@ public CdiSpiDecorator(ServletContextHandler context) throws UnsupportedOperatio
         try
         {
             Class<?> cdiClass = classLoader.loadClass("jakarta.enterprise.inject.spi.CDI");
+            Class<?> beanClass = classLoader.loadClass("jakarta.enterprise.inject.spi.Bean");
             Class<?> beanManagerClass = classLoader.loadClass("jakarta.enterprise.inject.spi.BeanManager");
             Class<?> annotatedTypeClass = classLoader.loadClass("jakarta.enterprise.inject.spi.AnnotatedType");
             Class<?> injectionTargetClass = classLoader.loadClass("jakarta.enterprise.inject.spi.InjectionTarget");
+            Class<?> injectionTargetFactoryClass = classLoader.loadClass("jakarta.enterprise.inject.spi.InjectionTargetFactory");
             Class<?> creationalContextClass = classLoader.loadClass("jakarta.enterprise.context.spi.CreationalContext");
             Class<?> contextualClass = classLoader.loadClass("jakarta.enterprise.context.spi.Contextual");
 
-            MethodHandles.Lookup lookup = MethodHandles.lookup();
-            _current = lookup.findStatic(cdiClass, "current", MethodType.methodType(cdiClass));
-            _getBeanManager = lookup.findVirtual(cdiClass, "getBeanManager", MethodType.methodType(beanManagerClass));
-            _createAnnotatedType = lookup.findVirtual(beanManagerClass, "createAnnotatedType", MethodType.methodType(annotatedTypeClass, Class.class));
-            _createInjectionTarget = lookup.findVirtual(beanManagerClass, "createInjectionTarget", MethodType.methodType(injectionTargetClass, annotatedTypeClass));
-            _createCreationalContext = lookup.findVirtual(beanManagerClass, "createCreationalContext", MethodType.methodType(creationalContextClass, contextualClass));
-            _inject = lookup.findVirtual(injectionTargetClass, "inject", MethodType.methodType(Void.TYPE, Object.class, creationalContextClass));
-            _dispose = lookup.findVirtual(injectionTargetClass, "dispose", MethodType.methodType(Void.TYPE, Object.class));
-            _release = lookup.findVirtual(creationalContextClass, "release", MethodType.methodType(Void.TYPE));
+            _current = cdiClass.getMethod("current", null);
+            _getBeanManager = cdiClass.getMethod("getBeanManager", null);
+            _createAnnotatedType = beanManagerClass.getMethod("createAnnotatedType", Class.class);
+            _getInjectionTargetFactory = beanManagerClass.getMethod("getInjectionTargetFactory", annotatedTypeClass);
+            _createInjectionTarget = injectionTargetFactoryClass.getMethod("createInjectionTarget", beanClass);
+            _createCreationalContext = beanManagerClass.getMethod("createCreationalContext", contextualClass);
+            _inject = injectionTargetClass.getMethod("inject", Object.class, creationalContextClass);
+            _dispose = injectionTargetClass.getMethod("dispose", Object.class);
+            _release = creationalContextClass.getMethod("release", null);
         }
         catch (Exception e)
         {
```

Comment:
I wonder if we should catch Throwable here an then if debug is on, dump the classloader and if any of the classes can be loaded and if so from where.